plugins/browser-plugin-performance-navigation-timing/src/contexts.ts

```diff
@@ -0,0 +1,83 @@
+import { PERFORMANCE_NAVIGATION_TIMING_SCHEMA } from './schemata';
+
+declare global {
+  interface Window {
+    mozPerformance: any;
+    msPerformance: any;
+    webkitPerformance: any;
+  }
+}
+
+/**
+ * Creates a context from the PerformanceNavigationTiming object
+ *
+ * @returns object PerformanceNavigationTiming context
+ */
+export function getPerformanceNavigationTimingContext() {
+  const performanceAlias =
+    window.performance || window.mozPerformance || window.msPerformance || window.webkitPerformance || {};
+
+  if (performanceAlias.getEntriesByType) {
+    const [performanceNavigationTiming] = performanceAlias.getEntriesByType('navigation') as [
+      PerformanceNavigationTiming
+    ];
```

Comment:
Nit: technically `PerformanceNavigationTiming[]` that just happens to always have a single element.